Source file encoding specifier support for template

AoiMoe opened the pull request:
Here is a small patch to add encoding handling support for template.

### How to test

1. save as ~/.config/rebar3/templates/00latin1.template in Latin-1:
```
%% coding:latin-1
{description, "±¼"}.
```
2. save as ~/.config/rebar3/templates/00utf8.template in UTF-8:
```
%% coding:utf-8
{description, "こんにちは"}.
```
3. run `rebar3 new`
```
$ ./rebar3 new
00latin1 (custom): ±¼
00utf8 (custom): こんにちは
...
```

ferd commented:
I get why that would be desirable. However I think for this to be acceptable we need:

1. Tests (see https://github.com/erlang/rebar3/blob/master/CONTRIBUTING.md)
2. To follow the Erlang character set encoding rules defined at https://erlang.org/doc/reference_manual/character_set.html#source-file-encoding

Part 1 is simply not done; Part 2 appears to be done for templates, though I'm not sure why the shell module has been modified there. I'd revert that one since the defaults in the string utils module should ensure it keeps working.

Let me know if you'd need help writing the tests and figuring that one out.

AoiMoe commented:
Sorry, I committed changes with --amend and pushed with -f by mistake.
I restored the last reviewed commit from reflog and pushed it forcibly again.

AoiMoe commented:
> Part 1 is simply not done

I added simple tests for rebar_templater:consult_template/3.

> Part 2 appears to be done for templates, though I'm not sure why the shell module has been modified there. I'd revert that one since the defaults in the string utils module should ensure it keeps working.

Okay, I just reverted it and applied simplified changes.

Thanks for your comments.
